### PR TITLE
coprocessor: RPN function UnaryNot

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

## What have you changed? (mandatory)

This PR adds UnaryNot RPN function.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

New unit test.

### CODE REVIEW

File: src/coprocessor/dag/rpn_expr/impl_op.rs [hunk 1]
```diff
@@ -54,6 +54,21 @@ impl RpnFnLogicalOr {
...
+#[derive(Debug, Clone, Copy, RpnFunction)]
+#[rpn_function(args = 1)]
+pub struct RpnFnUnaryNot;
+
+impl RpnFnUnaryNot {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg: &Option<i64>,
```
Comment (kennytm) [inline src/coprocessor/dag/rpn_expr/impl_op.rs]: TiDB currently is incompatible with MySQL 8 when decimal and floating point numbers is involved. If this function requires an `i64` input it is still easy for TiDB to fix the compatibility bug?

MySQL 8:

```
mysql> select not -0.01, not -0.0, not 0.0, not 0.01;
+-----------+----------+---------+----------+
| not -0.01 | not -0.0 | not 0.0 | not 0.01 |
+-----------+----------+---------+----------+
|         0 |        1 |       1 |        0 |
+-----------+----------+---------+----------+
1 row in set (0.01 sec)
```

On TiDB:

```
mysql> select not -0.01, not -0.0, not 0.0, not 0.01;
+-----------+----------+---------+----------+
| not -0.01 | not -0.0 | not 0.0 | not 0.01 |
+-----------+----------+---------+----------+
|         1 |        1 |       1 |        1 |
+-----------+----------+---------+----------+
1 row in set (0.00 sec)
```

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/impl_op.rs]: To handle this, maybe TiDB can introduce a new function called UnaryNotReal?

**Pull request merged**